test/python/test_ibmqjob_states.py

```diff
@@ -251,18 +258,39 @@ def test_only_final_states_cause_datailed_request(self):
 
                 with mock.patch.object(self._current_api, 'get_job',
                                        wraps=self._current_api.get_job):
-                    _ = job.status
+                    _ = job.status()
                     if status in API_FINAL_STATES:
                         self.assertTrue(self._current_api.get_job.called)
                     else:
                         self.assertFalse(self._current_api.get_job.called)
 
+    def test_job_has_finished_but_api_throws_temporarily(self):
+        """There could be temporary problems with the API server, but doesn't
+        mean the job has failed. The server could start working fine again, and
+        we should be able to retrieve the job"""
+
+        job = self.run_with_api(TemporaryThrowingServerButJobFinishedAPI())
+        job._wait_for_submission()
```

Comment:
Don't use an internal for waiting until initialization. Instead, use `self.wait_for_initialization`.

Reply:
Yeah, this is the first thing what I thought, but unfortunately `wait_for_initialization()` doesn't fit here (it calls `job.status()` for example). I just want to make sure that I have an ID, so wrapping  this internal in a function like:
```
def wait_for_submission():
     job._wait_for_submission()
```
feels like overkilling, right?